### ISSUE TITLE
Multiple not dismissible modal

### ISSUE THREAD

**Issue by mradzinski** [Dogfalo/materialize#1021]
If you try to open a new not-dismissible modal from a previously opened modal you loose the ability to interact with the original modal once the new modal get's dismissed.

**Comment by ciencia** [comment]
Opening a second modal once a modal is already opened doesn't work well. The overlay is referenced by the id "lean-overlay", so only one overlay can exist on the page. If you open a second modal, the overlay flashes, and it gets deleted when the second modal is closed, so you end with the first modal without overlay.

For now I only need 2 modals opened at the same time in one specific situation, where the second modal is created and populated in JS, so I've solved it by extending the lean-overlay styles into a new class, renaming the id of the lean-overlay of the element before the second modal is opened, and renaming it back to lean-overlay once the second modal is closed

**Comment by viniciusmelquiades** [comment]
I'm having this problem too, and I noticed a div with the id "lean-overlay" is created every time a new modal is opened.

I identified the problem after looking at the code in the file `leanModal.js`. The new overlay is created with jQuery, stored in a variable, and appended to the body. So far so good. 

The problem is, in lines **21**, **36** and **43**, for example, instead of using the variable stored for the overlay, they run a new selector, and since it's an id, it gets **only** the first overlay it finds (jQuery seems to work like this), and reanimates it, and add the click handler to it, if it's dismissible. Not only that, but the newly created overlay remains on the document, even after the modals have closed, because they cannot be referenced.

So the issue can be fixed just by changing `$("#lean-overlay")` to `overlay ` on the lines I said above.

Closing the modal also has the same problem, but it can be fixed storing the overlay variable into the modal data with the `.data` function of jQuery, and retrieving it when closing the modal.

I would create a pull-request, but the last one I submitted hasn't even seen interaction yet, so I won't waste my time unless one of the maintainers ask me to do this.
